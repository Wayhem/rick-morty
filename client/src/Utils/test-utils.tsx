import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { store } from 'Store'
import { theme } from 'Config/styled'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: FC = ({ children }: { children?: any }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  </ThemeProvider>
)

AllTheProviders.defaultProps = {
  children: <></>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult<Queries, HTMLElement> => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
