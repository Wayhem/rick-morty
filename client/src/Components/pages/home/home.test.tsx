import React from 'react'
import Home from 'Components/pages/home'
import { render } from 'Utils/test-utils'

test('Home renders and matches snapshot', () => {
  const { container } = render(<Home />)
  expect(container).toMatchSnapshot()
})
