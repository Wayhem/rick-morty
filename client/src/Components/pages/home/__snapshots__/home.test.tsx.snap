// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-gsTCUz dAGyOL"
  >
    <button
      class="sc-eCssSg flCiaF"
    >
      Log out
    </button>
    <div
      class="sc-hKgILt jfFCxP"
    >
      <button
        class="sc-eCssSg iOgwtR"
        color="#647183"
      >
        Go to favorites
      </button>
    </div>
    <div
      class="sc-dlfnbm epLfUN"
    >
      Hello, 
      
    </div>
  </div>
  <div
    class="sc-bdfBwQ lWayl"
  />
</div>
`;
